### DISCUSSION TITLE
Linking table rows to a new page

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [ToolJet/ToolJet#3869] [Q&A]
I have used different app builders in the past such as Budibase and appsmith....

These app builders by default allow for separate pages with different information.
I am assuming this is what the `tabs` widget offers?

What I was wondering as well is... when I select a table row... rather than use a modal widget, I would like a separate page/tab to open with the information from the selected row.
Also... Within the new page/tab have separate tabs shown, one for displaying the info, the other for editing the info and another for showing linked information

Is this possible?

**Comment by withshubh** [comment]
Solved on slack.

**Reply by tomasMizera** [reply]
How did this get solved please?

**Comment by karan-rathod-316** [comment]
Hi @tomasMizera,

To achieve this functionality, follow these steps:

1. Use the `Row clicked` event on the Table to trigger a `Switch page` action, which will navigate to a new page.
2. To display data from the Table on the new page, use the Table's [selectedRow](https://docs.tooljet.com/docs/widgets/table/table-csa-and-variables#exposed-variables) variable. This variable will hold the data from the selected row, which you can then use to populate the components on the new page.

I hope this answers your question. Feel free to post your queries on [slack](https://tooljet.slack.com/) for prompt response :-).

**Reply by tomasMizera** [reply]
Hi @karan-rathod-316, thanks for your reply! What I am looking for is to open a page in a new window (I did not explain it properly). At the moment we use `Open URL` action and do some hacks to alter href properly :)

**Reply by karan-rathod-316** [reply]
Hi @tomasMizera, I tried looking at other workarounds, but I think there's no way around it other than using the `Open URL` action currently. I also tried using local storage to save and retrieve the data on a new page, but it doesn't seem to be an optimal option.
